### PR TITLE
OAK-9967 : handled case to send lease update time in case of exception

### PR DESCRIPTION
**Pull request by rishabhdaim**

### CODE REVIEW

**Comment by rishabhdaim** [comment]
@joerghoh @mreutegg @stefan-egli could you please review and provide feedback?

File: oak-store-document/src/test/java/org/apache/jackrabbit/oak/plugins/document/DocumentNodeStoreTest.java [hunk 1]
```diff
@@ -413,6 +418,63 @@ public void throttlingWrapperCreatedWhenThrottlingIsEnabled_2() {
...
+    public void renewClusterIdLeaseTrue() throws IllegalAccessException {
+        DocumentStore documentStore = new MemoryDocumentStore();
+
+        DocumentNodeStoreStatsCollector statsCollector = mock(DocumentNodeStoreStatsCollector.class);
+        doNothing().when(statsCollector).doneLeaseUpdate(anyLong());
+
+        ClusterNodeInfo clusterNodeInfo = mock(ClusterNodeInfo.class);
+        when(clusterNodeInfo.renewLease()).thenReturn(true);
+
+        DocumentNodeStore nodeStore = builderProvider.newBuilder().setNodeStoreStatsCollector(statsCollector)
```
Comment (mreutegg) [inline oak-store-document/src/test/java/org/apache/jackrabbit/oak/plugins/document/DocumentNodeStoreTest.java:432]: Please use the `DocumentMKBuilderProvider` `builderProvider` to create a `DocumentNodeStore`. The `DocumentMKBuilderProvider` rule ensures the `DocumentNodeStore` instance is disposed after the test. Same for all other tests in this PR.

**Review by mreutegg — CHANGES_REQUESTED** [review]

File: oak-store-document/src/test/java/org/apache/jackrabbit/oak/plugins/document/DocumentNodeStoreTest.java (hunk 1 above)
Comment (rishabhdaim) [reply oak-store-document/src/test/java/org/apache/jackrabbit/oak/plugins/document/DocumentNodeStoreTest.java:432]: @mreutegg I have indeed used the `builderProvider` of `DocumentMKBuilderProvider`.

Comment (mreutegg) [reply oak-store-document/src/test/java/org/apache/jackrabbit/oak/plugins/document/DocumentNodeStoreTest.java:432]: Oops, you are right. Looks good then.

**Review by mreutegg — APPROVED** [review]

**Comment by mreutegg** [comment]
I triggered a rerun of the build on Jenkins.

**Pull request merged**